pymc/backends/base.py

```diff
@@ -204,10 +204,14 @@ def get_sampler_stats(self, stat_name, sampler_idx=None, burn=0, thin=1):
         vals = np.stack(
             [self._get_sampler_stats(stat_name, i, burn, thin) for i in sampler_idxs], axis=-1
         )
+
         if vals.shape[-1] == 1:
-            return vals[..., 0]
-        else:
-            return vals
+            vals = vals[..., 0]
+
+        if isinstance(vals, np.ndarray) and isinstance(vals[0], np.ndarray):
+            vals = np.vstack(vals)
```

Comment:
Due to the `vals.shape` above it's safe to assume `isinstance(vals, np.ndarray)`.

The `isinstance(vals[0], np.ndarray)` is also true for any 2D array. We'll have to check the dtype instead.

```suggestion
        if vals.dtype == np.dtype(object):
            try:
                vals = np.vstack(vals)
            except ValueError:
                # Most likely due to non-identical shapes.
                # Just stick with the object-array.
                pass
```